File: content/devto/2026-03/devto-publish.js
const https = require('https');
const fs = require('fs');

// Read the article content
const bodyMarkdown = fs.readFileSync(process.argv[2], 'utf8');

const apiKey = process.argv[3];

const payload = JSON.stringify({
  article: {
    title: "I Tried to Build a Decentralized AI Agent Marketplace. Here's What Broke (and What Worked)",
    body_markdown: bodyMarkdown,
    published: true,
    tags: ["ai", "agents", "blockchain", "typescript", "opensource"]
  }
});

const options = {
  hostname: 'dev.to',
  path: '/api/articles',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'api-key': apiKey,
    'Content-Length': Buffer.byteLength(payload)
  }
};

const req = https.request(options, (res) => {
  let data = '';
  res.on('data', (chunk) => data += chunk);
  res.on('end', () => {
    console.log('Status:', res.statusCode);
    console.log('Response:', data);
    try {
      const json = JSON.parse(data);
      if (json.url) {
        console.log('\nArticle URL:', json.url);
      }
    } catch (e) {}
  });
});

req.on('error', (e) => console.error('Error:', e.message));
req.write(payload);
req.end();

File: content/devto/2026-03/devto_publish.py
import requests
import json

# Read the article content
with open("article.md", "r", encoding="utf-8") as f:
    body_markdown = f.read()

# Dev.to API request
url = "https://dev.to/api/articles"
headers = {
    "Content-Type": "application/json",
    "api-key": "HThAUfdnNZPfreGabMMC9soa"
}
payload = {
    "article": {
        "title": "I Tried to Build a Decentralized AI Agent Marketplace. Here's What Broke (and What Worked)",
        "body_markdown": body_markdown,
        "published": True,
        "tags": ["ai", "agents", "blockchain", "typescript", "opensource"]
    }
}

response = requests.post(url, headers=headers, json=payload)
print(f"Status: {response.status_code}")
print(json.dumps(response.json(), indent=2))
